### PR TITLE
Use getters from the C API; return additional info from the file

### PR DESCRIPTION
**Pull request by dmbates**
I have taken the liberty of extending the `ReadStatDataFrame` while converting the code to use some of the getter functions defined in the C API.

If you prefer a clean PR that only switches to using the getter functions, I will prepare one.

I planned to include the value labels as well as these are important in working with categorical variables.  Strangely, these file types often represent categorical data with a small number of categories as `Float64`.  Conversion to, say, CategoricalArray's requires the value labels.

### CODE REVIEW

File: src/ReadStat.jl [hunk 1]
```diff
@@ -12,9 +12,9 @@ end
 ##
 ##############################################################################
 
-import DataValues: DataValueVector, DataValueArray
+using DataValues.DataValueVector
```
Comment (davidanthoff) [inline src/ReadStat.jl]: I'm always confused about these, but isn't ``import Packagename: thing`` the recommended style?

File: src/ReadStat.jl [hunk 2]
```diff
@@ -108,71 +82,120 @@ end
...
+end
 
-    data_type = get_type(variable)
-    jtype = Float64
+function get_type(var::Ptr{Void})
+    data_type = ccall((:readstat_variable_get_type, libreadstat), Cint, (Ptr{Void},), var)
+   
     if data_type == READSTAT_TYPE_STRING
-        jtype = String
+        String
```
Comment (davidanthoff) [inline src/ReadStat.jl]: Could you add ``return`` statements here? I think that would make it easier to read this part of the code.

File: src/ReadStat.jl (hunk 1 above)
Comment (dmbates) [reply src/ReadStat.jl]: My understanding is that `import` is used if you want to extend a definition, say by adding a method to a generic.  There is a [manual section](https://docs.julialang.org/en/stable/manual/faq/#What-is-the-difference-between-%22using%22-and-%22importall%22?-1) about `using` versus `importall`.  I think that "using" just ensures that the package is loaded and allow you to use unqualified names for symbols that are exported from the package.

File: src/ReadStat.jl (hunk 2 above)
Comment (dmbates) [reply src/ReadStat.jl]: Will do.

**Comment by davidanthoff** [comment]
Closing this in favor of #36.

**Pull request closed without merging**